syft/core/hooks/torch/guard.py

```diff
@@ -61,11 +59,29 @@ def types_guard(self, torch_type_str):
                 "Tried to receive a non-Torch object of type {}.".format(
                     torch_type_str))
 
-    def tensor_contents_guard(self, contents):
+
+    def tensor_contents_guard(self,contents):
         """tensor_contents_guard(contents) -> contents
-        TODO: check to make sure the incoming list isn't dangerous to use for
+        check to make sure the incoming list isn't dangerous to use for
                constructing a tensor (likely non-trivial). Accepts the list of JSON objects
-               and returns the list of JSON ojects. Should throw and exception if there's a
+               and returns the list of JSON ojects.Throws an exception if there's a
                security concern.
         """
+        
+        is_instance = lambda tensor : all(isinstance(digit, (int, float)) for digit in tensor)
+        assert type(contents) in (list,tuple) , "A list of JSON objects is expected"
+        for json_object in contents : 
+            try :
+                json_object = json.loads(json_object)
+            except Exception as e:
```

Comment:
Please get rid of this before merging -- better to raise the exception if someone feeds it a bad json object.  If this is meant to be used for duck-typing, it's probably best to figure out a way to differentiate the specific cases you're trying to handle.  Maybe this means catching specific Exception types -- for example, if it's not a JSON object, pass and continue to the contents guard logic below, but if it _is_ a JSON object and it's just not loadable with `json.loads`, propagate that exception up.  This way, the function won't silently fail.

You can also switch out the try/except for an if/elif/else block.  If that's more readable, that's likely best, although sometimes the try/except duck-type style is easier to read/understand.

Reply:
Yup, trying this.